### ISSUE TITLE
lintr mistakenly identifies commented code

### ISSUE THREAD

**Issue by cboettig** [jimhester/lintr#286]
lintr thinks that 

```r
## ------------------------------------------------------------------------
```

is commented-out code, and thus flags it as an issue.  When you convert an RMarkdown `.Rmd` to `.R` with `knitr::purl` you get such lines separating each chunk.

**Comment by topspinj** [comment]
I have a similar issue. In my function documentation, I provide a url that has a `#` in its path. 

my code: 

```
#' @references http://scikit-learn.org/stable/modules/model_evaluation.html#r2-score-the-coefficient-of-determination
```

lintr detects the `#` as commented code:

```
Warning “Commented code should be removed”
```

**Comment by mattyb** [comment]
example:
```
## author: John + Jane
```

**Comment by jimhester** [comment]
In the current master neither of the first two result in lints, the third does, but I don't see any way to prevent that, that code is valid R code. I think this is as fixed as we can make it.
